packages/headless/src/features/search/search-slice.ts

```diff
@@ -8,7 +8,13 @@ function handleRejectedSearch(
   state: SearchState,
   action: ReturnType<SearchAction['rejected']>
 ) {
-  state.error = action.payload ? action.payload : null;
+  const hasError = action.payload ? action.payload : null;
```

Comment:
I don't think a switch case is necessary, we get null when we abort the request

Reply:
I'll check for the accessToken renewal, that could be a case where we dont want to clear the results

Reply:
Tested and works well for Token renewal, the hasError remains false

Reply:
Would `error` be a better name for the variable?

```
const error = action.payload || null;
```